packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTScrollViewComponentView.mm

```diff
@@ -530,9 +530,8 @@ - (BOOL)_shouldDisableScrollInteraction
   return NO;
 }
 
-- (ScrollViewEventEmitter::Metrics)_scrollViewMetrics
+- (void)_setScrollViewMetrics:(ScrollViewEventEmitter::Metrics &)metrics
```

Comment:
this approach is not really iOS-sy...

What do you feel about having two methods: `_scrollViewMetrics` and `_scrollViewMetricsWithVelocity:(CGPoint)velocity andTargetContentOffset:(CGPoint)targetContentOffset`?

The first one, behaves as before. The second one does:
```
auto metrics = [self __scrollViewMetrics];
metrics.targetContentOffset.x = targetContentOffset->x;
metrics.targetContentOffset.y = targetContentOffset->y;
metrics.velocity.x = velocity.x;
metrics.velocity.y = velocity.y;
return metrics;
```

This should also minimize the changes in `RCTScrollViewComponentView`.